src/pykiso/lib/auxiliaries/zephyr.py

```diff
@@ -0,0 +1,212 @@
+
+        :param twister_path: Path to the twister tool
+        :param test_directory: The directory to search for the Zephyr test project
+        :param testcase_name: The name of the Zephyr test
+        :param wait_for_start: Wait for Zyephyr test start
+
+        """
+        super().__init__(
+            is_proxy_capable=True, tx_task_on=False, rx_task_on=False, **kwargs
+        )
+        self.test_directory = test_directory
+        self.test_name = test_name
+        self.wait_for_start = wait_for_start
+        self.channel = com
+        self.running = False
+
+    def _run_command(self, cmd_message: str, cmd_data: bytes = None) -> bool:
+        return True
+
+    def _receive_message(self, timeout_in_s: float) -> None:
+        pass
+
+    def start_test(
+        self, test_directory: Optional[str] = None, test_name: Optional[str] = None
+    ) -> None:
+        """Start the Zephyr test
+
+        :param test_directory: The directory to search for the Zephyr test project. Defaults to the test_directory from YAML.
+        :param testcase_name: The name of the Zephyr test. Defaults to the testcase_name from YAML.
+        """
+        test_directory = (
+            test_directory if test_directory is not None else self.test_directory
+        )
+        test_name = test_name if test_name is not None else self.test_name
+        if test_directory is None:
+            raise ZephyrError("test_directory parameter is not set.")
+        if test_name is None:
+            raise ZephyrError("test_name parameter is not set.")
+
+        if self.running:
+            raise ZephyrError("Twister test is already running")
+
+        self.outdir = str(Path(test_directory).resolve() / "twister-out")
+        logging.internal_info(f"Using Twister output directory: {self.outdir}")
+        self.channel.cc_send(
+            {
+                "command": "start",
+                "args": [
+                    "-vv",
+                    "-T",
+                    test_directory,
+                    "-s",
+                    test_name,
+                    "--outdir",
+                    self.outdir,
+                    "--report-dir",
+                    self.outdir,
+                ],
+            }
+        )
+
+        self.running = True
+        if self.wait_for_start:
+            # Read twister output until test case was started
+            while True:
+                line = self._readline()
+                if line is None:
+                    return
+
+                if "OUTPUT: START - " in line:
+                    log.internal_debug("Zephyr test started.")
+                    return
+
+    def _readline(self) -> Optional[str]:
+        """Read a line from the twister process
+
+        :return: The line as string or None if the process has finished
+        """
+        if not self.running:
+            return None
+
+        while True:
+            msg = self.channel.cc_receive(timeout=None)["msg"]
```

Comment:
```suggestion
            msg = self.channel.cc_receive(timeout=None).get("msg")
```
Better be safe than sorry ;)

Reply:
Done